From oliver@neukum.name Sat Oct 28 09:07:35 2006
Date: Sat, 28 Oct 2006 18:07:25 +0200 (MEST)
From: Oliver Neukum <oliver@neukum.name>
To: linux-usb-devel@lists.sourceforge.net, greg@kroah.com, <vojtech@suse.cz>
Subject: USB: usblp: fix system suspend for some systems
Message-Id: <200610281807.45846.oliver@neukum.name>

this has been confirmed to fix suspend problems with usblp.

Signed-off-by: Oliver Neukum <oliver@neukum.name>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/class/usblp.c |    2 --
 1 file changed, 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/class/usblp.c
+++ gregkh-2.6/drivers/usb/class/usblp.c
@@ -1203,8 +1203,6 @@ static int usblp_suspend (struct usb_int
 	down (&usblp->sem);
 	/* we take no more IO */
 	usblp->sleeping = 1;
-	/* we wait for anything printing */
-	wait_event (usblp->wait, usblp->wcomplete || !usblp->present);
 	usblp_unlink_urbs(usblp);
 	up (&usblp->sem);
 	mutex_unlock (&usblp_mutex);
